### ISSUE TITLE
No automatic rollback when migration fails

### ISSUE THREAD

**Issue by Gyalomalom** [doctrine/DoctrineBundle#1703]
Hello,

If an exception is encountered while running a migration, all changes up to that point are kept instead of being rolled back. It makes it difficult to run the migration again after fixing the issue. Are there any plans to fix this?

**Comment by ostrolucky** [comment]
This bundle has no relation to migrations, you are probably looking for https://github.com/doctrine/DoctrineMigrationsBundle But I also have to say that MySQL doesn't support tranasctions for table changes, so I don't think there is anything that can be done here.

**Comment by stof** [comment]
doctrine/migrations already supports wrapping migrations in a transaction for platforms that support it.
Personally, I consider this a big winning point of Postgresql over MySQL (I remember my pain years ago when I was using MySQL and needed to recover from failed migration, and I definitely don't regret that time).
For MySQL (and other platforms that don't support DDL in transactions), there is nothing we can do in Doctrine.

**Comment by greg0ire** [comment]
More on this at https://www.doctrine-project.org/projects/doctrine-migrations/en/stable/explanation/implicit-commits.html

TL;DR: stop using MySQL
